src/metrics/core.rs

```diff
@@ -0,0 +1,125 @@
+use std::sync::atomic::{AtomicBool, Ordering};
+
+use once_cell::sync::Lazy;
+use prometheus_client::{encoding::text::encode, registry::Registry};
+
+use crate::metrics::iroh;
+
+pub(crate) static CORE: Lazy<Core> = Lazy::new(Core::default);
+
+pub(crate) struct Core {
+    enabled: AtomicBool,
+    registry: Registry,
+    iroh_metrics: iroh::Metrics,
+}
+
+impl Default for Core {
+    fn default() -> Self {
+        let mut reg = Registry::default();
+        Core {
+            enabled: AtomicBool::new(false),
+            iroh_metrics: iroh::Metrics::new(&mut reg),
+            registry: reg,
+        }
+    }
+}
+
+impl Core {
+    pub(crate) fn registry(&self) -> &Registry {
+        &self.registry
+    }
+
+    pub(crate) fn iroh_metrics(&self) -> &iroh::Metrics {
+        &self.iroh_metrics
+    }
+
+    pub(crate) fn encode(&self) -> Result<Vec<u8>, std::io::Error> {
+        let mut buf = vec![];
+        encode(&mut buf, self.registry())?;
```

Comment:
Rust is trying so hard to nerdsnipe me into wanting to overthink the allocations of this `Vec` here.  I'm trying to resist.  This is fine.